skbeam/core/recip.py

```diff
@@ -342,3 +342,42 @@ def gisaxs(incident_beam, reflected_beam, pixel_size, detector_size,
     qr = np.sqrt(qx**2 + qy**2)
 
     return gisaxs_output(alpha_i, theta_f, alpha_f, tilt_angle, qx, qy, qz, qr)
+
+
+def generate_q_bins(rmax, pixel_size, distance, wavelength, rmin=0):
+    """
+    Generate the Q bins at the resolution of the detector
+    Parameters
+    -----------
+    rmax: float
+        The maximum radial distance on the detector in distance units.
+        Note that this should go to the bottom edge of the pixel.
+    pixel_size: float
```

Comment:
which dimension of pixels?  Do we need to worry about non-square pixels?  Should the resolution be comptued using the height / width or the diagonal?